hathor/sysctl/p2p/manager.py

```diff
@@ -269,3 +286,56 @@ def refresh_auto_hostname(self) -> None:
     def reload_entrypoints_and_connections(self) -> None:
         """Kill all connections and reload entrypoints from the peer config file."""
         self.connections.reload_entrypoints_and_connections()
+
+    def get_whitelist(self) -> str:
+        """Get status of current whitelist."""
+        if self.connections.peers_whitelist:
+            wl_object = self.connections.peers_whitelist
+            return get_whitelist_msg(wl_object)
+
+        return 'No whitelist.'
+
+    def set_whitelist(self, new_whitelist: str) -> None:
+        """Set the whitelist-only mode. If 'on' or 'off', simply changes the
+        following status of current whitelist. If an URL of Filepath, changes
+        the whitelist object, following it by default.
+        It does not support eliminating the whitelist (passing None)."""
+
+        wl_object: URLPeersWhitelist | FilePeersWhitelist
+        wl_url = urlparse(new_whitelist)
+        option: str = new_whitelist.lower().strip()
+        if os.path.isfile(new_whitelist):
+            # Fetch the peerIds in the file
+            # Must check whether the peerIds within make sense.
+            # Must start the looping call of the object.
+            wl_object = FilePeersWhitelist(self.connections.reactor, new_whitelist)
+
+        elif wl_url.scheme == 'https' and wl_url.netloc:
+            # Fetch the peerIds in the URL
+            # Must ?? check whether they make sense?
+            # Must start the looping call of the object.
+            wl_object = URLPeersWhitelist(self.connections.reactor, new_whitelist, True)
+
+        elif option in ('on', 'off'):
+            # Set the whitelist tracking ON or OFF for the currently given whitelist.
+            if option == "on":
+                # Turning the whitelist on immediately blocks all connections.
+                if self.connections.peers_whitelist:
+                    self.connections.peers_whitelist.follow_wl()
+                self.connections.whitelist_toggle(True)
+                return
+            else:
+                # Turning the whitelist off will update in the next refresh cycle
+                if self.connections.peers_whitelist:
+                    self.connections.peers_whitelist.unfollow_wl()
+                self.connections.whitelist_toggle(False)
+                return
```

Comment:
Here you can use the same factory you used in `cli_builder.py`. It avoids duplicated code and giver you a unified syntax for whitelisting.